packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts

```diff
@@ -33,7 +32,7 @@ export const processProtectedData = async ({
   args,
   inputFiles,
   secrets,
-  workerpool,
+  workerpool, // if the user want to specify all workerpool he should use Null_Address
```

Comment:
this will not work unless you also change the `fetchWorkerpoolOrderbook` call

- https://api.market.iex.ec/workerpoolorders?chainId=134&workerpool=0x0000000000000000000000000000000000000000 will always return an empty orderbook
- https://api.market.iex.ec/workerpoolorders?chainId=134&workerpool=any will return orders for any workerpools

```ts
    const workerpoolOrderbook = await iexec.orderbook.fetchWorkerpoolOrderbook({
      workerpool: vWorkerpool === ethers.ZeroAddress ? 'any' : vWorkerpool, // if address zero was chosen use any workerpool
      app: vApp,
      dataset: vProtectedData,
      minTag: SCONE_TAG,
      maxTag: SCONE_TAG,
});
```

Reply:
yeah it's true. 
Fix : Commit: a2bb109c04ff5025bc62bda5961e76210b6782e2
Add test : Commit: d92cf2975da9bc4081a402fba8d5fcb1e86a2b70

PS : I don't succeed to mock the getResultFromCompletedTask function import under the hood by processProtectedData. It's the same issue as consumeProtectedData.

Reply:
will do in another PR